leshan-core/src/main/java/org/eclipse/leshan/senml/json/jackson/SenMLJsonJacksonEncoderDecoder.java

```diff
@@ -75,7 +75,7 @@ public SenMLPack fromSenML(byte[] jsonString) throws SenMLException {
             }
             return new SenMLPack(serDes.deserialize(node.iterator()));
 
-        } catch (JsonException | IOException e) {
+        } catch (JsonException | IllegalArgumentException | IOException e) {
```

Comment:
(If we encapsulate `IllegalArgumentException` in  `JsonException`   we should revert this change)